### ISSUE TITLE
Support for Gradle 8

### ISSUE THREAD

**Issue by gourgouris** [google/webcrypto.dart#84]
hi, I am getting this error when I import the plugin and try to run Android in the emulator (the other targets work)

```
* What went wrong:
A problem occurred configuring project ':webcrypto'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.
```

which from what I remember a couple of months back, indicates that the plugin does not have namespace specified in their build.gradle - it is a requirement for Gradle 8 for all subprojects.

Without this, I dont I have any workaround other than downgrading gradle, which is something I'd like to avoid.

My gradle dependency: com.android.tools.build:gradle:8.2.0

**Comment by jonasfj** [comment]
@gourgouris reproduction steps please?

We have integration tests for android:
https://github.com/google/webcrypto.dart/blob/5e6d20f820531d2b7b05935c1d78f38a036035e8/.github/workflows/test.yml#L117-L139

Could we reproduce this by trying to build the `example/` app in release mode?

Do we need to upgrade the `example/` app to reproduce this?

I could probably use some help here.


--------------------------

What are the changes required here? (happy to take PRs, ping me on any PRs)

Will those changes have implications on people using an older gradle version? (if so this needs to be documented in the `CHANGELOG` before we ship any changes).

**Comment by gourgouris** [comment]
hi @jonasfj, can't really test it right now, but my thinking is the following:

replication steps
1. in example/
2. in android/build.gradle change classpath 'com.android.tools.build:gradle:7.1.2' to 'com.android.tools.build:gradle:8.2.2' 
3. in android/gradle/wrapper/gradle-wrapper.properties change distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-all.zip to distributionUrl=https\://services.gradle.org/distributions/gradle-8.2-all.zip

compiling now should give you the same error.

To fix it, the first I'd do would be to add a namespace line in the android/build.gradle of the plugin AND the example (eg like here: https://github.com/flutter/packages/blob/0af905d779d52fc81ffc239c211e8c5956154f34/packages/path_provider/path_provider_android/android/build.gradle#L27)
And that should theoritically get rid of the error

**Comment by jonasfj** [comment]
Okay, still unclear if it'll affect people using older versions of gradle.

I see new flutter templates have `android/build.gradle` with:
```
    if (project.android.hasProperty("namespace")) {
        namespace 'com.example.mypkg'
    }
```

We should probably investigate this further.

**Comment by gourgouris** [comment]
Yes, this is because older versions of gradle (<4.2) do not support the namespace property and thus would fail to build (unknown property 'namespace').

From the offically maintained flutter packages, I can understand this is the way to go now.
Relevant PRs:
https://github.com/flutter/packages/commit/6284c2d4e46a5d289e77cb03a9457543b97f750b
(bulk adds namespace to all packages)

https://github.com/flutter/packages/commit/a86beafa8912609275225847198c9c0164957d09
(bulk conditionalizes namespace, 3 days apart)

My point is it looks safe, since all flutter packages do it anyway, but that's as far I can help I think.
